mongoengine/base/document.py

```diff
@@ -628,7 +628,7 @@ def _delta(self):
                 del set_data['_id']
 
         # Determine if any changed items were actually unset.
-        for path, value in set_data.items():
+        for path, value in list(set_data.items()):
```

Comment:
converting to list could be bad for memory?

Reply:
yeah I can't change it because the code inside the for loop is actually altering the `set_data` dict and that will raise an error under python3. Personally I don't expect it to leak much so I'm not sure that its worth spending effort on optimizing this.

Under Python3, there is no regression with the current version of mongoengine as 2to3 used to turn the code to this exact line behind the scenes.
Under python2, we have a regression as it will actually create 2 lists, I'll improve this by [using](https://python-future.org/compatible_idioms.html#dict-keys-values-items-as-a-list) `from future.utils import listitems`